File: static/js/scrums.js
$(function () {
  const teamPageId = $('#team-page-id').val();
  const currentUserId = $('#current-user-id').val();

  // 등록
  $('#scrum-register-btn').on('click', function () {
    const content = $('#scrum-content-input').val().trim();
    if (!content) {
      alert('내용을 입력해주세요');
      return;
    }

    $.ajax({
      url: `/api/team_pages/${teamPageId}/scrums`,
      method: 'POST',
      contentType: 'application/json',
      data: JSON.stringify({ content: content }),
      success: function (res) {
        const s = res.scrum;
        const html = `
          <li class="scrum-item box py-3 px-4 mb-2" data-id="${s._id}" data-user-id="${currentUserId}">
            <span class="scrum-date tag is-light">${s.log_date}</span>
            <span class="scrum-name has-text-weight-semibold ml-2">${s.user_name}</span>
            <span class="scrum-content ml-2"></span>
            <div class="buttons is-pulled-right">
              <button class="scrum-edit-btn button is-small is-link is-light">수정</button>
              <button class="scrum-delete-btn button is-small is-danger is-light">삭제</button>
            </div>
          </li>
        `;
        const $li = $(html);
        $li.find('.scrum-content').text(s.content); // XSS 방지: text()로 안전하게 삽입
        $('#scrum-list').prepend($li);  // 리스트 맨 위에 추가
        $('#scrum-content-input').val('');
      },
      error: function (xhr) {
        alert(xhr.responseJSON.error);
      }
    });
  });

  // 수정 버튼 클릭 → 인라인 입력창으로 전환
  $('#scrum-list').on('click', '.scrum-edit-btn', function () { //이벤트 위임
    const $li = $(this).closest('.scrum-item');
    const currentContent = $li.find('.scrum-content').text();

    const $input = $('<input type="text" class="scrum-edit-input input is-small">');
    $input.val(currentContent); // XSS 방지: val()로 안전하게 삽입
    $li.find('.scrum-content').replaceWith($input);

    $(this).text('저장')
      .removeClass('scrum-edit-btn is-link is-light')
      .addClass('scrum-save-btn is-primary');
  });

  // 저장 버튼 클릭 → PATCH 요청
  $('#scrum-list').on('click', '.scrum-save-btn', function () {
    const $li = $(this).closest('.scrum-item');
    const scrumId = $li.data('id');
    const newContent = $li.find('.scrum-edit-input').val().trim();

    if (!newContent) {
      alert('내용을 입력해주세요');
      return;
    }

    $.ajax({
      url: `/api/scrums/${scrumId}`,
      method: 'PATCH',
      contentType: 'application/json',
      data: JSON.stringify({ content: newContent }),
      success: function (res) {
        const $span = $('<span class="scrum-content ml-2"></span>').text(res.content);
        $li.find('.scrum-edit-input').replaceWith($span);
        $li.find('.scrum-save-btn').text('수정')
          .removeClass('scrum-save-btn is-primary')
          .addClass('scrum-edit-btn is-link is-light');
      },
      error: function (xhr) {
        alert(xhr.responseJSON.error);
      }
    });
  });

  // 삭제
  $('#scrum-list').on('click', '.scrum-delete-btn', function () {
    const $li = $(this).closest('.scrum-item');
    const scrumId = $li.data('id');

    if (!confirm('삭제하시겠습니까?')) return;

    $.ajax({
      url: `/api/scrums/${scrumId}`,
      method: 'DELETE',
      success: function () {
        $li.remove();
      },
      error: function (xhr) {
        alert(xhr.responseJSON.error);
      }
    });
  });

  // 더보기
  $('#scrum-more-btn').on('click', function () {
      $('.scrum-hidden').removeClass('scrum-hidden');
      $(this).hide();
  });
});

File: routes/team/scrums.py
from flask import Blueprint, request, jsonify
from bson import ObjectId
from datetime import datetime, date
from config import db

scrums_bp = Blueprint('scrums', __name__)

@scrums_bp.route("/api/team_pages/<team_page_id>/scrums", methods=["POST"])
def add_scrums(team_page_id):
    current_user_id = ObjectId('6a8d72d4cd0d6b3be61313b7') #임시로 하드코딩, 추후 나중에 JWT에서 꺼내올 부분
    user = db.users.find_one({"_id": current_user_id})
    user_name = user["name"] if user else "알수없음"

    data = request.json # 파이썬의 딕셔너리로 반환 
    content = data.get("content", "").strip() 
    if not content:
        return jsonify({"error": "내용을 입력해주세요"}), 400 

    today = date.today().isoformat() # "2026-08-25" 형태

    result = db.scrums.insert_one({
        "team_page_id": ObjectId(team_page_id),
        "user_id": current_user_id,
        "content": content,
        "log_date": today,
        "created_at": datetime.now()
    })

    return jsonify({
        "success": True,
        "scrum":{
            "_id": str(result.inserted_id),
            "content": content,
            "log_date": today,
            "user_name": user_name,
        }
    })


@scrums_bp.route("/api/scrums/<scrum_id>", methods=["PATCH"])
def update_scrums(scrum_id):
    current_user_id = ObjectId('6a8d72d4cd0d6b3be61313b7') #임시로 하드코딩, 추후 나중에 JWT에서 꺼내올 부분
  
    scrum = db.scrums.find_one({"_id": ObjectId(scrum_id)})
    if not scrum:
        return jsonify({"error":"존재하지 않는 스크럼입니다."}), 404 

    # 본인 글인지 검증 
    if scrum["user_id"] != current_user_id:
        return jsonify({"error":"본인 글만 수정할 수 있습니다."}), 403

    data = request.json
    content = data.get("content", "").strip()

    if not data:
        return jsonify({"error":"내용을 입력해주세요"}), 403

    db.scrums.update_one(
        {"_id": ObjectId(scrum_id)}, # 첫 번째 인자: "어떤 문서를" 찾을지 (조건)
        {"$set": {"content": content}} # 두 번째 인자: "뭘 어떻게" 바꿀지 (변경 내용)
    )

    return jsonify({"success": True, "content": content})


@scrums_bp.route("/api/scrums/<scrum_id>", methods=["DELETE"])
def delete_scrums(scrum_id):
    current_user_id = ObjectId('6a8d72d4cd0d6b3be61313b7') #임시로 하드코딩, 추후 나중에 JWT에서 꺼내올 부분

    scrum = db.scrums.find_one({"_id": ObjectId(scrum_id)})

    if not scrum:
        return jsonify({"error":"존재하지 않는 스크럼입니다."}), 404 

    # 본인 글인지 검증 
    if scrum["user_id"] != current_user_id:
        return jsonify({"error":"본인 글만 삭제할 수 있습니다."}), 403

    db.scrums.delete_one({"_id": ObjectId(scrum_id)})

    return jsonify({"success": True})